### DISCUSSION TITLE
No payloads being sent with persitent_menu

### DISCUSSION THREAD

**Discussion by chadcomsci** [iTeam-S/Ampalibe#106] [Q&A] [answered]
Hi Ampalibe team, first off this is a really nice tool you've got here, I've been using it for testing the messenger platform for a few weeks now and it works without a hitch

But it just stopped working out of the blue as of today: none of my persistent_menu payloads are being sent now, there was no change to the code and tokens that would lead to this error

Could you point out how this can be debugged? no event seems to be fired from clicking on my buttons with defined payloads anymore

edit: even the example codes with payloads don't work anymore, but sending the payload directly, like "/do_something" via chat works fine and reaches my webhook

Many thanks

**Comment by gaetan1903** [comment]
Hi,

so the message is sent as a simple messge? it is not redirected to the target payload?

it is surely a bug of the client platform used.

P.S: If you use `Messenger Lite`, this application does not support postback messages,

Try to test on different client platform namely Facebook web, Facebook Messenger, Facebook Lite latest version.

**Reply by chadcomsci** [reply]
Oh, I'm using full messenger app on Android and messenger.com, both have the same behavior

I depend entirely on the persistent_menu options to do the chat flow, it worked for the last few weeks until now

sending plaintext chat like "hello" works fine and is received by my server, it triggers the proper wildcard '/' command endpoint. Sending in the command directly, like '/eat' also works and triggers the right commands

but clicking on my persistent menu options does nothing at all, it doesn't seem to send anything out and nothing is received on my server

I even tried with the examples given on the documentation, anything to test for payload transmission, but nothing seems to be sent from the messenger side and nothing reaches my API..

Strange because this was working just a few days ago, but nothing seems to indicate any issue

edit: I tried moving to the LTS version from Alpha, and checked on other browsers, cleaned cache, changed tokens and everything, still the same thing..

edit2: if it serves as a clue, the before_receive event isn't triggered at all. messages are just left at 'sent' status.

**Reply by gaetan1903** [reply]
can you confirm that `messages_postback` is enabled on subcriptions fields on developer.facebook ? 
![image](https://github.com/iTeam-S/Ampalibe/assets/43904633/b71ac8e6-3579-4ec5-aef7-670eacc9fbcf)

**Reply by chadcomsci** [reply]
> can you confirm that `messages_postback` is enabled on subcriptions fields on developer.facebook ? 
> ![image](https://github.com/iTeam-S/Ampalibe/assets/43904633/b71ac8e6-3579-4ec5-aef7-670eacc9fbcf)
> 

Yes, I checked and postback is still enabled..

**Comment by gaetan1903** [comment]
Can you provide a video to show click and terminal console, and before receive code, and make a print in this code.

**Reply by gaetan1903** [reply]
@chadcomsci  how about it

**Reply by chadcomsci** [reply]
to be clear:

a. record devtool > console during click on payload item (menu click)
b. print out text within before_receive()

is that right?

I've tried both of those initially, but there is nothing being displayed on the dev-tool > console (browser side) nor uvicorn, after clicking on a persistent_menu item.

as for the before_receive event, it does not reach that either..

in any case, I will try it again now for peace of mind

**Reply by chadcomsci** [reply]
@gaetan1903 

thanks for the help btw

Here's the vid of sending payloads and sending normal text:
https://github.com/iTeam-S/Ampalibe/assets/18280559/6ab2372c-6b6a-461c-9243-c7da42825176

Here's what reaches the server side:
![image](https://github.com/iTeam-S/Ampalibe/assets/18280559/8783fca2-5226-4835-a6fa-3e057f13c9d8)

it's supposed to print something upon receipt, but nothing is inside:
![image](https://github.com/iTeam-S/Ampalibe/assets/18280559/cbebdaac-2110-4892-b119-d491116b0b1e)

nothing suspicious under the network transmission:
![image](https://github.com/iTeam-S/Ampalibe/assets/18280559/db920de1-37a8-4fe9-9807-49cde4c42ed0)

I'm not sure how to debug it further when I can't see anything happening, what else could I try?

**Comment by gaetan1903** [comment]
```python
import ampalibe
from ampalibe.ui import QuickReply
from ampalibe import Payload, Messenger

chat = Messenger()


@ampalibe.command('/')
def main(sender_id, cmd, **ext):
    quick_rep = [
        QuickReply(
            title='Angela',
            payload=Payload('/membre', name='Angela', ref='2016-sac')
        ),
        QuickReply(
            title='Rivo',
            payload=Payload('/membre', name='Rivo')
        )
    ]
    chat.send_quick_reply(sender_id, quick_rep, 'Who?')


@ampalibe.command('/membre')
def get_membre(sender_id, cmd, name, **ext):
    chat.send_text(sender_id, "Hello " + name)

    # if the arg is not defined in the list of parameters,
    # it is put in the extends variable
    if ext.get('ref'):
        chat.send_text(sender_id, 'your ref is ' + ext.get('ref'))

``` 

can you try this code and verif if it works when clicking button, 






can you also verify if there is a error log here
![image](https://github.com/iTeam-S/Ampalibe/assets/43904633/c4538e0f-00d2-4e84-9895-2a49e9674463)

**Reply by salts-project** [reply]
(oops sorry this is still me ^_^;)

ok, uploaded that code and here are the responses

a. sent a message to trigger '/' command, quick actions appear successfully

![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/4f82ac5c-ec60-487e-a12e-31dad1baeb1c)
![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/4c0c4215-189e-4867-abd1-f6c5aeab3ac3)

b. clicking on the button

![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/e4b41b53-bef7-412e-bc97-892718d6c3b1)
![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/b005e1ec-6556-4a39-b659-0617870f8915)

Payload seems to go through for quick replies..

But look at the response here for the example using the advanced payload mechanisms (this is what I tested earlier):

![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/3a145317-be77-4b07-8e29-32fb6cc1f874)

Only the wildcard command '/' was recognized and no payload was delivered after even clicking separate buttons:

![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/3cc8df93-3ffe-40d2-9c9b-22a7e004cd19)

![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/25b1ade3-6b09-41a6-a0be-450a9d55eb8f)

it's the same behavior as the persistent menu

Nothing is in the error log as well:

![image](https://github.com/iTeam-S/Ampalibe/assets/19583135/fe32001a-e170-4ac7-a4ea-b2a3e4e03da6)

**Comment by gaetan1903** [comment]
I tried on my side and I did not encounter any problem


https://github.com/iTeam-S/Ampalibe/assets/43904633/560abbbb-15f6-4740-8d67-d459bc15b291


contact me on discord and we can view together your problem user: @gaetan1903

**Comment by chadcomsci** [comment, marked as answer]
We figured this one out, thanks to @gaetan1903

Seems the Facebook page itself started bugging out and stopped sending payloads altogether, with no indication of any errors.

It was fixed by linking to a new/different page, using the same app as before.

**Comment by gaetan1903** [comment]
removing link to the app then re-adding it fixes it
